kubestash/__init__.py

```diff
@@ -661,25 +679,34 @@ def cmd_daemonall(args):
             print("checking DynamoDB Stream for changes...")
 
         response = client.get_records(ShardIterator=shard_iterator, Limit=100)
-        print(len(response['Records']))
 
-        if len(response['Records']) > 0:
-            for record in response['Records']:
-                key = record['dynamodb']['Keys']['name']['S']
-                if args.verbose:
-                    print("detected DynamoDB changes, running push command...")
-                argscopy = copy.copy(args)
-                argscopy.secretname = key
-                cmd_pushall(argscopy)
+        if 'Records' not in response:
+            print ("Error: No Records found in response for Shard:{0}".format(shard_iterator))
 
-        time.sleep(args.interval)
+        else:
+            records = response['Records']
 
+            if args.verbose:
+                print "Shard ID:{0}, Record Len:{1}".format(shard_iterator, len(records))
+
+            if len(records) > 0:
+                for record in records:
+                    try:
+                        key = record['dynamodb']['Keys']['name']['S']
+                        if args.verbose:
+                            print("detected DynamoDB changes, running push command...")
+                        argscopy = copy.copy(args)
+                        argscopy.secretname = key
+                        cmd_pushall(argscopy)
```

Comment:
Why are we still doing a pushall? I thought the idea was to run a selective patch on the secret.ENV for every change.

Reply:
@captn3m0 need some clarity on this. Lets sync up when am back from indore